### PR TITLE
[E2E] Run OSS-specific tests on PR

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- It adds a small subset of OSS-specific tests to the PR check

#### Why (some background):
We decided to optimize this workflow by run E2E tests only against EE version, as it is a superset of the OSS version. However, there is a crucial subset of tests (usually negative testing) that we absolutely must run against OSS version. The challenge was how to achieve this without either duplicating the whole `e2e-tests` job or spinning up X containers for each of the E2E groups (folders).

Some conditional logic was needed in the job matrix, after which we could add `cypress-grep` plugin.

### Important
We have to tag OSS-specific tests with the `@OSS` tag. It works on `describe`, `context` and `it`.

### Caveats
Cypress-grep plugin [doesn't work with parameterized test names](https://github.com/cypress-io/cypress-grep/issues/109). That means that we cannot use `describeOSS` any more, nor we can construct such tests dynamically. For example:
```js
["A", "B"].forEach(name => {
  it(`test #{name}`, { tags: "@OSS" }, ()=> {
    // cypress-grep would not "see" this
  });
});
```
1. There is an effort to fix this
2. Even if we have to work around this, the number of OSS-specific tests is fairly small and we can always write them in such a way that the plugin works on them

### Note
After we merge this PR it should be safe to remove any group that we currently run on PR using GHA from Circle CI checks! 🎉 🍾 🥳

### CODE REVIEW

File: frontend/test/metabase/scenarios/sharing/subscriptions.cy.spec.js [hunk 1]
```diff
@@ -273,7 +273,7 @@ describe("scenarios > dashboard > subscriptions", () => {
     });
   });
 
-  describe("OSS email subscriptions", () => {
+  describe("OSS email subscriptions", { tags: "@OSS" }, () => {
```
Comment (diogormendes) [inline frontend/test/metabase/scenarios/sharing/subscriptions.cy.spec.js:276]: Is this one working with 2 OSS to parse ?

Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/sharing/subscriptions.cy.spec.js:276]: @diogormendes not sure exactly what you mean?

Does it parse the "OSS" in "OSS email subscriptions"?

**Comment by diogormendes** [comment]
nit, only 242ms, but do you know why [pivot_tables.cy.spec.js](https://github.com/metabase/metabase/runs/6632100580?check_suite_focus=true#step:10:467) is running since there was no change ?

**Comment by nemanjaglumac** [comment]
> nit, only 242ms, but do you know why [pivot_tables.cy.spec.js](https://github.com/metabase/metabase/runs/6632100580?check_suite_focus=true#step:10:467) is running since there was no change ?

For whatever reason, Cypress grep cannot determine whether that file has grep tags or not so it adds it to the queue just in case. This is the related log:
https://github.com/metabase/metabase/runs/6632417085?check_suite_focus=true#step:10:255

```
Could not determine test names in file: metabase/scenarios/visualizations/pivot_tables.cy.spec.js
Will run it to let the grep filter the tests
```

**Comment by nemanjaglumac** [comment]
One more thing to note:
We will be now using 2 additiona machines for **every PR**!

One is used to build `OSS` uberjar and the other one to run `OSS` tests.

cc @ariya @diogormendes

**Review by flamber — APPROVED** [review]
This looks good and I would say fairly manageable to maintain and write.
grepping is still a little new to me, but the runs in the PR looks correct.

**Pull request merged**